syntax = "proto3";

package sudoku;

option go_package = "/sudoku";

service SudokuService {
  rpc Solve (SudokuRequest) returns (SudokuResponse);
  rpc GetSudokuList (GetSudokuRequest) returns (GetSudokuResponse);
  rpc CheckSudoku(CheckSudokuRequest) returns (CheckSudokuResponse);
}

message SudokuRequest {
  string puzzle = 1;
  bool isSteps = 2;
}

message SudokuResponse {
  string solution = 1;
  string error = 2;
}

message GetSudokuRequest {
  string token = 1;
}

message GetSudokuResponse {
  repeated int32 ids = 1;
  repeated string boards = 2;
  repeated int32 difficulties = 3;
  repeated bool isSolved = 4;
  string error = 5;
}

message CheckSudokuRequest {
  string solution = 1;
  string token = 2;
  int32 sudokuId = 3;
}

message CheckSudokuResponse {
  bool isCorrect = 1;
}